webapp/src/serviceworker.ts

```diff
@@ -197,12 +198,49 @@ function initWebappServiceWorker() {
     });
 
     self.addEventListener("fetch", (ev: ServiceWorkerEvent) => {
-        ev.respondWith(caches.match(ev.request)
-            .then(response => {
-                return response || fetch(ev.request)
-            }))
+        ev.respondWith(handleFetch(ev));
     });
 
+    async function handleFetch(ev: ServiceWorkerEvent): Promise<Response> {
+        if (ev.request.url.startsWith(pageUrl)) {
+            let path = ev.request.url.slice(pageUrl.length);
+            if (path.startsWith("/")) path = path.slice(1);
+
+            // If this is just the main page with a query parameter, attempt
+            // to fetch from the network and fall back to the cache if it fails
+            if (path.charAt(0) === "?") {
+                let response: Response;
+                try {
+                    response = await fetch(ev.request);
+
+                    // Store this response in the cache in case the user tries
+                    // to visit this same query param offline
```

Comment:
Just a clarifying question. This just means that because we are caching the site with query parameters, the user _can_ get to the site with those configurations, right? Like if the user removes the query param they will still be able to use the site without those added features from the query parameters because our base targets work offline?

Reply:
Oh this might be the answer already that you had in the PR comment:
>This PR caches responses when a query parameter is present and uses those as a fallback when we're offline. Note that this is only used as a fallback; we will still fetch live whenever possible.

Reply:
yup! if you look at the init function above, the first thing it caches is the main index.html with no query parameters